benchmark/benchmark.cpp

```diff
@@ -109,71 +102,67 @@ UMF_BENCHMARK_TEMPLATE_DEFINE(alloc_benchmark, scalable_pool_fix,
 
 UMF_BENCHMARK_REGISTER_F(alloc_benchmark, scalable_pool_uniform)
-    ->Args({10000, 0, 8, 64 * 1024, 8})
-    ->Threads(4)
-    ->Threads(1);
+    ->Apply(&default_alloc_uniform_size);
 #endif
 // Multiple allocs/free
+static void
+default_multiple_alloc_fix_size(benchmark::internal::Benchmark *benchmark) {
+    benchmark->Args({10000, 4096});
+    benchmark->Threads(4);
+    benchmark->Threads(1);
+}
+
+static void
+default_multiple_alloc_uniform_size(benchmark::internal::Benchmark *benchmark) {
+    benchmark->Args({10000, 8, 64 * 1024, 8});
+    benchmark->Threads(4);
+    benchmark->Threads(1);
+}
 
 UMF_BENCHMARK_TEMPLATE_DEFINE(multiple_malloc_free_benchmark, glibc_fix,
                               fixed_alloc_size, glibc_malloc);
 
 UMF_BENCHMARK_REGISTER_F(multiple_malloc_free_benchmark, glibc_fix)
-    ->Args({10000, 4096})
-    ->Threads(4)
-    ->Threads(1);
+    ->Apply(&default_multiple_alloc_fix_size);
 
 UMF_BENCHMARK_TEMPLATE_DEFINE(multiple_malloc_free_benchmark, glibc_uniform,
                               uniform_alloc_size, glibc_malloc);
 UMF_BENCHMARK_REGISTER_F(multiple_malloc_free_benchmark, glibc_uniform)
-    ->Args({10000, 8, 64 * 1024, 8})
-    ->Threads(4)
-    ->Threads(1);
+    ->Apply(&default_multiple_alloc_uniform_size);
 
 UMF_BENCHMARK_TEMPLATE_DEFINE(multiple_malloc_free_benchmark, proxy_pool,
                               fixed_alloc_size,
                               pool_allocator<proxy_pool<os_provider>>);
 
 UMF_BENCHMARK_REGISTER_F(multiple_malloc_free_benchmark, proxy_pool)
-    ->Args({10000, 4096})
-    ->Threads(4)
-    ->Threads(1);
+    ->Apply(&default_multiple_alloc_fix_size);
 
 UMF_BENCHMARK_TEMPLATE_DEFINE(multiple_malloc_free_benchmark, os_provider,
                               fixed_alloc_size,
                               provider_allocator<os_provider>);
 UMF_BENCHMARK_REGISTER_F(multiple_malloc_free_benchmark, os_provider)
-    ->Args({10000, 4096})
-    ->Threads(4)
-    ->Threads(1);
+    ->Apply(&default_multiple_alloc_fix_size);
 
 #ifdef UMF_POOL_DISJOINT_ENABLED
 UMF_BENCHMARK_TEMPLATE_DEFINE(multiple_malloc_free_benchmark, disjoint_pool_fix,
                               fixed_alloc_size,
                               pool_allocator<disjoint_pool<os_provider>>);
 UMF_BENCHMARK_REGISTER_F(multiple_malloc_free_benchmark, disjoint_pool_fix)
-    ->Args({10000, 4096})
-    ->Threads(4)
-    ->Threads(1);
+    ->Apply(&default_multiple_alloc_fix_size);
 
 // TODO: debug why this crashes
 /*UMF_BENCHMARK_TEMPLATE_DEFINE(multiple_malloc_free_benchmark,
                               disjoint_pool_uniform, uniform_alloc_size,
                               pool_allocator<disjoint_pool<os_provider>>);
 UMF_BENCHMARK_REGISTER_F(multiple_malloc_free_benchmark, disjoint_pool_uniform)
-    ->Args({10000, 0, 8, 64 * 1024, 8})
-    ->Threads(4)
-    ->Threads(1);
+    ->Apply(&default_multiple_alloc_uniform_size);
```

Comment:
soo, now this one is either wrongly copy-pasted (and should be `default_alloc_uniform_size`), or previously it was wrongly set...?

Reply:
Now is correct